### PR TITLE
Branch new delete command

### PR DESCRIPTION
**Pull request by yuhaochua**
Add assertion in DeleteCommand#prepareDelete() and JUnit tests for DeleteCommand. It is now ensured that users can only delete an equipment by specifying serial number and show relevant error messages if anything other attributes are provided.

### CODE REVIEW

File: src/main/java/seedu/command/DeleteCommand.java [hunk 1]
```diff
@@ -32,15 +36,23 @@ public DeleteCommand(ArrayList<String> commandStrings) {
...
     public CommandResult execute() {
         String equipmentName;
+        String serialNumber;
+
         try {
-            equipmentName = equipmentManager.getEquipmentList().get(commandStrings.get(0)).getItemName();
+            serialNumber = prepareDelete();
+            if (serialNumber.equals("")) {
+                return new CommandResult(ONLY_SN_ACCEPTED);
+            }
```
Comment (shunyao643) [inline src/main/java/seedu/command/DeleteCommand.java]: You might need to adjust this based on the logic in the other comment further down.

File: src/main/java/seedu/command/DeleteCommand.java [hunk 2]
```diff
@@ -54,4 +66,20 @@ public boolean equals(Object other) {
...
+        // the case where delimiterPos = -1 is impossible as
+        // ARGUMENT_FORMAT and ARGUMENT_TRAILING_FORMAT regex requires a '/'
+        assert delimiterPos != -1 : "Each args will need to include minimally a '/' to split arg and value upon";
+        String argType = argString.substring(0, delimiterPos);
+        String argValue = argString.substring(delimiterPos + 1);
+        assert argType.equals("s") : ONLY_SN_ACCEPTED;
+        if (argType.equals("s")) {
+            return argValue;
+        }
+        return "";
```
Comment (shunyao643) [inline src/main/java/seedu/command/DeleteCommand.java]: Line 83 is unreachable if you assert that `argType.equals("s")` in line 79. I do think the assertion is a good idea though, so maybe you'd like to just go directly to `return argValue` after line 79?

File: src/test/java/seedu/command/DeleteCommandTest.java [hunk 3]
```diff
@@ -41,4 +41,33 @@ void execute_validSerialNumber_success() {
...
+    @Test
+    void execute_notSerialNumber_fail() {
+        deleteCommand = new DeleteCommand(new ArrayList<>(
+                Arrays.asList("n/Speaker3")
+        ));
+        deleteCommand.setEquipmentManager(new EquipmentManager());
+
+        CommandResult actualResult = deleteCommand.execute();
+        CommandResult expectedResult =
+                new CommandResult("Only serial number accepted for deleting equipment");
```
Comment (shunyao643) [inline src/test/java/seedu/command/DeleteCommandTest.java]: Want to use the `ONLY_SN_ACCEPTED` string in `DeleteCommand` you previously defined?

**Review by shunyao643 — CHANGES_REQUESTED** [review]
Thanks for putting in the assertions! Only some minor changes requested.

File: src/test/java/seedu/command/DeleteCommandTest.java (hunk 3 above)
Comment (yuhaochua) [reply src/test/java/seedu/command/DeleteCommandTest.java]: Okay, will change this to `ONLY_SN_ACCEPTED`!

File: src/main/java/seedu/command/DeleteCommand.java (hunk 2 above)
Comment (yuhaochua) [reply src/main/java/seedu/command/DeleteCommand.java]: Yes that makes sense, will make the changes accordingly.

**Review by shunyao643 — APPROVED** [review]
LGTM!

**Pull request merged**